homeassistant/components/synologydsm/sensor.py

```diff
@@ -112,6 +112,7 @@ def run_setup(event):
         monitored_conditions = config.get(CONF_MONITORED_CONDITIONS)
 
         api = SynoApi(host, port, username, password, unit, use_ssl)
+        api.update()
```

Comment:
It would be good to rewrite this set up function. Delaying adding the sensors is not the correct approach. We should just delay the first api update until home assistant start.

There are more problems too, like try... except inside api init method.

Reply:
Indeed, that would be a nice integration to "fix" entirely :smile: 

An exemple : I've tried to configure multiple time before finding the right host and port for the sensor --> no log / Exception to tell me that it fails and the sensors where created with an unknown state ...

Actually, it was delaying, now on start.


I'll rewrite the integration using config flow, and a better error handling (PR next week).
Don't know if we can merge this fix until I work on it ?

PS : All this work just by a quick look for the issue #30629 :sweat_smile:

Reply:
We call `add_entities` with second argument `True` which makes entity update method be called during entity addition. This method will call the throttled api update method. So this change here shouldn't be needed.

Reply:
Although it looks like an api update is needed to populate the data that is used to create some of the sensors.

Reply:
It is, otherwise all sensors are in None state

Reply:
No, this is not the problem, I think. I've looked into the library and there are requests done already on instantiation of the api.

I found a bug though:
This retry should return the response, but currently doesn't return anything:
https://github.com/StaticCube/python-synology/blob/599064b5ddbf1f215ae03d85434865efe1cbf504/SynologyDSM/SynologyDSM.py#L471

Reply:
I think we shouldn't make this change but do the rewrite to clean this up thoroughly.

Also would be good to if the bug in the library is fixed.

Reply:
OK, will check the API and refactor the integration.

Reply:
PR opened : https://github.com/StaticCube/python-synology/pull/10